src/Components/SingleArticle.jsx

```diff
@@ -16,28 +16,50 @@ import Typography from '@mui/material/Typography';
 
 export default function SingleArticle({singleArticle}){
   //im not using singleArticle
-    const [article, SetArticle] = useState([])
+    const [article, setArticle] = useState([])
     const [isLoading, setIsLoading] = useState(true)
     const [isError, setIsError] = useState(false)
     const [showComments, setShowComments] = useState(false)
     const [comments, setComments] = useState([])
+    const [voteCount, setVoteCount] = useState(null)
+    const [hasVoted, setHasVoted] = useState(false)
     const {article_id} = useParams()
+    
 
     useEffect(()=>{
         getSingleArticle(article_id)
         .then((response)=>{
-            SetArticle(response.article)
+            setArticle(response.article)
+            setVoteCount(response.article.votes)
             setIsLoading(false)
         })
         .catch(()=>{
           setIsError(true)
         })
-    },[])
+    },[voteCount])
 
     function HandleCommentClick() {
       setShowComments(!showComments)
     }
 
+    function handleVote(votes){
+      const newVotes = {inc_votes: votes}
+      setVoteCount((voteCount)=>{voteCount + votes})
+      patchVotes(article_id, newVotes)
+      
+    }
+
+    function handleUpvote(){
+      setHasVoted(true)
+      handleVote(1)
+    }
+
+    function handleDownvote(){
+      setHasVoted(true)
+      handleVote(-1)
+      
```

Comment:
You could potentially make this cleaner by just calling your handleVote function in the onClick attribute in the tag, you could grab the 1 or -1 value from a value attribute in the tag or by calling the function with an argument. Just makes things a bit more DRY.